src/reducers/requests/users.js

```diff
@@ -148,6 +148,57 @@ function getUser(state: RequestStatusType = initialRequestState(), action: Gener
     );
 }
 
+/* ******************************************************************************
+ * getUserAnalytics                                                        */ /**
+ *
+ * Reducer for getUserAnalytics. Because the getUserAnalytics is parameterized
+ * by a name which determines what is actually being requested and returned,
+ * the status for the particular named analytics needs to be tracked.
+ *
+ * The request state is in a property with the analytic's name as the key, and
+ * that is why the initial state is an empty object instead of initialRequestState()
+ *
+ * @example
+ * if you request the name 'UserInteractions', and it succeeds, the state will
+ * look as follows:
+ * getUserAnalytics: {
+ *    UserInteractions: {
+ *        status: success,
+ *        error: null,
+ *    }
+ * }
+ *
+ * @param {RequestStatusType} state - containing RequestStatusType objects for every user analytics request
+ * @param {GenericAction} action - containing data for each action
+ *
+ * @returns {object} containing the new requests.users.getUserAnalytics state
+ */
+function getUserAnalytics(state: Object = {}, action: GenericAction): Object {
+    const HANDLED_ACTION_TYPES = [
+        UserTypes.GET_USER_ANALYTICS_REQUEST,
+        UserTypes.GET_USER_ANALYTICS_SUCCESS,
+        UserTypes.GET_USER_ANALYTICS_FAILURE,
+    ];
+
+    // We only want to create request state if the action is a GET_USER_ANALYTICS request action
+    // so we check for an action type we want to handle up front.
+    if (!HANDLED_ACTION_TYPES.some((type) => type === action.type)) {
+        return state;
+    }
+
+    // Use the standard handleRequest helper to update the named analytics request state
+    // property.
+    const curUserAnalyticsNameState = state[action.data.name] || initialRequestState();
+    return {
+        ...state,
+        [action.data.name]: handleRequest(
+            ...HANDLED_ACTION_TYPES,
+            curUserAnalyticsNameState,
+            action
+        ),
+    };
+}
+
```

Comment:
At first I wondered why is this not simply:
```
function getUserAnalytics(state: RequestStatusType = initialRequestState(), action: GenericAction): RequestStatusType {
    return handleRequest(
        UserTypes.GET_USER_ANALYTICS_REQUEST,
        UserTypes.GET_USER_ANALYTICS_SUCCESS,
        UserTypes.GET_USER_ANALYTICS_FAILURE,
        state,
        action
    );
}
```
But I realized that you were (correctly) handling requests for different named user analytics.

However, the function still defined the state parameter as a RequestStatusType, and returned a RequestStatusType, and I still thought the function could make use of the helper `handleRequest` function.
As that is more complicated to deal with I wrote and committed how I though this should be implemented which is what is above, and I'm opening this comment for any discussion of the change

Reply:
I really like what you did here! I think the @param shouldn't be 'RequestStatusType' any more, since it's just an object, but everything else looks really concise. And it is probably good to use their handleRequest function.

Reply:
Oh you're totally right about the param. That's what happens when I'm not paying attention while fixing conflicts when rebasing. I had written the comment before you changed the comment style w/o any params or return. When I saw you had added those descriptions, I just copied and pasted w/o carefully reading it.

Thanks for your feedback, and for giving it a quick test.

Reply:
Ok fixed that.